apollo-router/src/plugins/cors.rs

```diff
@@ -0,0 +1,1083 @@
+                http::HeaderValue::from_static("true"),
+            );
+        }
+
+        // Set Access-Control-Allow-Headers (only for preflight requests)
+        if is_preflight {
+            if !allow_headers.is_empty() {
+                // Join the headers with commas for a single header value
+                let header_value = allow_headers.join(", ");
+                response.headers_mut().insert(
+                    ACCESS_CONTROL_ALLOW_HEADERS,
+                    http::HeaderValue::from_str(&header_value)
+                        .unwrap_or_else(|_| http::HeaderValue::from_static("")),
+                );
+            } else {
+                // If no headers are configured, mirror the client's Access-Control-Request-Headers
+                if let Some(request_headers) = request_headers {
+                    if let Ok(headers_str) = request_headers.to_str() {
+                        response.headers_mut().insert(
+                            ACCESS_CONTROL_ALLOW_HEADERS,
+                            http::HeaderValue::from_str(headers_str)
+                                .unwrap_or_else(|_| http::HeaderValue::from_static("")),
+                        );
+                    }
+                }
+            }
+        }
+
+        // Set Access-Control-Expose-Headers (only for non-preflight requests)
+        if !is_preflight {
+            if let Some(headers) = expose_headers {
+                // Join the headers with commas for a single header value
+                let header_value = headers.join(", ");
+                response.headers_mut().insert(
+                    ACCESS_CONTROL_EXPOSE_HEADERS,
+                    http::HeaderValue::from_str(&header_value)
+                        .unwrap_or_else(|_| http::HeaderValue::from_static("")),
+                );
+            }
+        }
+
+        // Set Access-Control-Allow-Methods (for preflight requests)
+        if is_preflight {
+            // Join the methods with commas for a single header value
+            let method_value = methods.join(", ");
+            response.headers_mut().insert(
+                ACCESS_CONTROL_ALLOW_METHODS,
+                http::HeaderValue::from_str(&method_value)
+                    .unwrap_or_else(|_| http::HeaderValue::from_static("")),
+            );
+        }
+
+        // Set Access-Control-Max-Age (only for preflight requests)
+        if is_preflight {
+            if let Some(max_age) = max_age {
+                let max_age_secs = max_age.as_secs();
+                response.headers_mut().insert(
+                    ACCESS_CONTROL_MAX_AGE,
+                    http::HeaderValue::from_str(&max_age_secs.to_string())
+                        .unwrap_or_else(|_| http::HeaderValue::from_static("")),
+                );
+            }
+        }
+
+        // Set Vary header - append to existing values instead of overwriting
+        Self::append_vary_header(response, "Origin");
```

Comment:
I just added a couple vary system tests.  I ended up getting responses with:
```Vary: origin, Origin```  with this branch.  

It appears this needs to conditionally append `origin` only if it doesnt already exist / hasnt been appended by any other Router code/coprocessors/rhai/plugins (and do a case insensitive check for it already existing)